modules/caddyhttp/headers/caddyfile.go

```diff
@@ -76,62 +74,68 @@ func parseCaddyfile(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error)
 				replacement = h.Val()
 			}
 			err := applyHeaderOp(
-				hdr.Response.HeaderOps,
-				hdr.Response,
+				handler.Response.HeaderOps,
+				handler.Response,
 				field,
 				value,
 				replacement,
 			)
 			if err != nil {
 				return nil, h.Err(err.Error())
 			}
-			handlers = append(handlers, hdr)
+			if len(handler.Response.HeaderOps.Delete) > 0 {
+				handler.Response.Deferred = true
+			}
 		}
 
 		// if not, they should be in a block
 		for h.NextBlock(0) {
-			hdr := Handler{
-				Response: &RespHeaderOps{
-					HeaderOps: &HeaderOps{},
-				},
-			}
-
 			field := h.Val()
 			if field == "defer" {
-				deferred = true
+				handler.Response.Deferred = true
 				continue
 			}
 			if hasArgs {
-				return nil, h.Err("cannot specify headers in both arguments and block")
+				return nil, h.Err("cannot specify headers in both arguments and block") // because it would be weird
 			}
+
+			// sometimes it is habitual for users to suffix a field name with a colon,
+			// as if they were writing a curl command or something; see
+			// https://caddy.community/t/v2-reverse-proxy-please-add-cors-example-to-the-docs/7349
```

Comment:
Just to be more specific, cause it's a long thread

```suggestion
			// https://caddy.community/t/v2-reverse-proxy-please-add-cors-example-to-the-docs/7349/19
```